packager/media/codecs/h264_byte_to_unit_stream_converter.cc

```diff
@@ -52,6 +69,32 @@ bool H264ByteToUnitStreamConverter::GetDecoderConfigurationRecord(
   buffer.AppendInt(static_cast<uint16_t>(last_pps_.size()));
   buffer.AppendVector(last_pps_);
 
+  // handle profile special cases, refer to ISO/IEC 14496-15 Section 5.3.3.1.2
+  uint8_t profile_indication = last_sps_[1];
+  if (profile_indication == 100 || profile_indication == 110 || 
+      profile_indication == 122 || profile_indication == 144) {
+      
+      const H264Sps* sps = RetrieveSps(last_sps_);
+      if (sps == nullptr)
+        return false;
+
+      uint8_t reserved_chroma_format = 0xff & (sps->chroma_format_idc);
```

Comment:
Sorry, I think I made a mistake in the previous suggestion. Not sure what I was thinking...

It should be `0xfc | sps->chroma->format_idc`;

Remove the unnecessary "()". Similar below.